src/scene/graphics/shared/svg/SVGToGraphicsPath.ts

```diff
@@ -1,244 +1,202 @@
-                lastY += data[dataIndex + 3];
-
-                dataIndex += 4;
+                lastX += data[2];
+                lastY += data[3];
                 break;
             case 'T':
-
-                lastX = data[dataIndex++];
-                lastY = data[dataIndex++];
+                lastX = data[0];
+                lastY = data[1];
 
                 path.quadraticCurveToShort(
                     lastX, lastY
                 );
-
                 break;
             case 't':
-
-                lastX += data[dataIndex++];
-                lastY += data[dataIndex++];
+                lastX += data[0];
+                lastY += data[1];
 
                 path.quadraticCurveToShort(
                     lastX, lastY
                 );
-
                 break;
             case 'A':
-
-                lastX = data[dataIndex + 5];
-                lastY = data[dataIndex + 6];
+                lastX = data[5];
+                lastY = data[6];
 
                 path.arcToSvg(
-                    data[dataIndex],
-                    data[dataIndex + 1],
-                    data[dataIndex + 2],
-                    data[dataIndex + 3],
-                    data[dataIndex + 4],
+                    data[0],
+                    data[1],
+                    data[2],
+                    data[3],
+                    data[4],
                     lastX, lastY
                 );
-
-                dataIndex += 7;
-
                 break;
             case 'a':
-                lastX += data[dataIndex + 5];
-                lastY += data[dataIndex + 6];
+                lastX += data[5];
+                lastY += data[6];
 
                 path.arcToSvg(
-                    data[dataIndex],
-                    data[dataIndex + 1],
-                    data[dataIndex + 2],
-                    data[dataIndex + 3],
-                    data[dataIndex + 4],
+                    data[0],
+                    data[1],
+                    data[2],
+                    data[3],
+                    data[4],
                     lastX, lastY
                 );
-
-                dataIndex += 7;
-
                 break;
             case 'Z':
             case 'z':
                 path.closePath();
+                if (subpaths.length > 0)
+                {
+                    currentSubPath = subpaths.pop();
+                    lastX = currentSubPath?.startX ?? 0;
+                    lastY = currentSubPath?.startY ?? 0;
+                }
+                currentSubPath = null;
                 break;
             default:
                 // #if _DEBUG
                 warn(`Unknown SVG path command: ${type}`);
                 // #endif
         }
+
+        if (type !== 'Z' && type !== 'z')
+        {
+            if (currentSubPath === null)
+            {
+                currentSubPath = { startX: lastX, startY: lastY };
+                subpaths.push({ ...currentSubPath });
```

Comment:
can we just do `subpaths.push(currentSubPath);` here? no need to destructure as its creating a new object.